packages/web/src/pages/purchases-page/PurchasesTable.tsx

```diff
@@ -0,0 +1,235 @@
+import { MouseEvent, useCallback, useMemo } from 'react'
+
+import {
+  accountSelectors,
+  formatUSDCWeiToUSDString,
+  SquareSizes,
+  statusIsNotFinalized,
+  USDCContentPurchaseType,
+  USDCPurchaseDetails,
+  useGetTrackById,
+  useGetUserById
+} from '@audius/common'
+import moment from 'moment'
+import { Cell, Row } from 'react-table'
+
+import DynamicImage from 'components/dynamic-image/DynamicImage'
+import { Table } from 'components/table'
+import { Text } from 'components/typography'
+import UserBadges from 'components/user-badges/UserBadges'
+import { useTrackCoverArt2 } from 'hooks/useTrackCoverArt'
+import { useSelector } from 'utils/reducer'
+
+import styles from './PurchasesTable.module.css'
+
+const { getUserId } = accountSelectors
+
+type PurchaseCell = Cell<USDCPurchaseDetails>
+type PurchaseRow = Row<USDCPurchaseDetails>
+
+export type PurchasesTableColumn =
+  | 'contentName'
+  | 'artist'
+  | 'date'
+  | 'value'
+  | 'spacerLeft'
+  | 'spacerRight'
+
+export type PurchasesTableSortMethod =
+  | 'contentId'
+  | 'sellerUserId'
+  | 'createdAt'
+export type PurchasesTableSortDirection = 'asc' | 'desc'
+
+type PurchasesTableProps = {
+  columns?: PurchasesTableColumn[]
+  data: USDCPurchaseDetails[]
+  isVirtualized?: boolean
+  loading?: boolean
+  onClickRow?: (txDetails: USDCPurchaseDetails, index: number) => void
+  onSort: (
+    sortMethod: PurchasesTableSortMethod,
+    sortDirection: PurchasesTableSortDirection
+  ) => void
+  fetchMore: (offset: number, limit: number) => void
+  totalRowCount?: number
+  scrollRef?: React.MutableRefObject<HTMLDivElement | undefined>
+  fetchBatchSize: number
+}
+
+const defaultColumns: PurchasesTableColumn[] = [
+  'spacerLeft',
+  'contentName',
+  'artist',
+  'date',
+  'value',
+  'spacerRight'
+]
+
+// TODO: When we support collection purchases
+const CollectionNameCell = ({ id }: { id: number }) => {
+  return <div />
+}
+
+const TrackNameCell = ({ id }: { id: number }) => {
+  const { status, data: track } = useGetTrackById({ id })
+  const image = useTrackCoverArt2(id, SquareSizes.SIZE_150_BY_150)
+  const loading = statusIsNotFinalized(status) || !track
+  return loading ? null : (
+    <div className={styles.contentName}>
+      <DynamicImage wrapperClassName={styles.artwork} image={image} />
+      <Text variant='body' size='small'>
+        {track.title}
+      </Text>
+    </div>
+  )
+}
+
+const UserNameAndBadge = ({ userId }: { userId: number }) => {
```

Comment:
nit: Badges (plural)

We should probably have a popover for the username

Also would this work? https://github.com/AudiusProject/audius-client/blob/62a0c72df7a0563e72f6db1b0bfe6ddc507735dd/packages/web/src/components/user-name-and-badges/UserNameAndBadges.tsx